pkg/controllers/cloud_config_sync_controller_test.go

```diff
@@ -241,8 +243,8 @@ var _ = Describe("Cloud config sync controller", func() {
 		for _, cm := range allCMs.Items {
 			Expect(cl.Delete(ctx, cm.DeepCopy(), deleteOptions)).To(Succeed())
 			Eventually(
-				apierrors.IsNotFound(cl.Get(ctx, client.ObjectKeyFromObject(cm.DeepCopy()), &corev1.ConfigMap{})),
-			).Should(BeTrue())
+				cl.Get(ctx, client.ObjectKeyFromObject(cm.DeepCopy()), &corev1.ConfigMap{}),
```

Comment:
This is a gotcha of using eventually statements, what you're doing here is calling `cl.Get` while constructing the `Eventually`, which means `Eventually` gets a single result, which never changes.

The original is also broken, but you have two choices here to fix it:
```suggestion
				func() error { return cl.Get(ctx, client.ObjectKeyFromObject(cm.DeepCopy()), &corev1.ConfigMap{})},
```
Or use [komega](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/envtest/komega) as we do in CPMS
```suggestion
				komega.Get(ctx, client.ObjectKeyFromObject(cm.DeepCopy()), &corev1.ConfigMap{}),
```
You'll need to either create a new instance of komega or set the global komega client before you can do this.